pkg/sys/unshare_linux.go

```diff
@@ -65,6 +65,9 @@ func UnshareAfterEnterUserns(uidMap, gidMap string, unshareFlags uintptr, f func
 	}
 
 	if pidfd == -1 || !SupportsPidFD() {
+		if pidfd != -1 {
```

Comment:
nit: Won't let me comment on line 66 where I want this as there's no diff, but given there's no branch where we won't want to close this, could we just `defer unix.Close(pidfd)`? I'm worried someone might change this function in the future and introduce a new branch where this goes unclosed again. If we don't support pidfd we'd just get EBADF, but if you want you could do the below all the same.

```go
defer func() {
   if pidfd != -1 { unix.Close(pidfd) }
}()
```

Reply:
@dcantah, I also wanted to make defer work for this since I agree with your concerns. The thing is, I'm not sure it'll work cleanly for the 2.0 release branch because of that dup fix:
  https://github.com/containerd/containerd/blob/release/2.0/pkg/sys/unshare_linux.go#L79-L87

  I can't tell if the original pointer stays open or if the bug fix closes it properly. Either way, though, I think using defer on main is cleaner, so I'm going to implement that regardless of how we handle the backport.